### PR TITLE
issue(913): Add runtime deprecation for unnamed Icon

### PR DESCRIPTION
**Pull request by pcln-barry**
https://github.com/priceline/design-system/issues/913

### CODE REVIEW

File: packages/icons/src/Icon.js [hunk 1]
```diff
@@ -5,14 +5,22 @@ import PropTypes from 'prop-types'
 import * as icons from './index'
 
 const BaseIcon = ({ name, title, desc, titleId, descId, ...props }) => {
+  const isDev = process.env.NODE_ENV !== 'production'
```
Comment (sdalonzo) [inline packages/icons/src/Icon.js:8]: I think this part should go into master now, to help folks work on this sooner.

File: packages/icons/src/Icon.js [hunk 2]
```diff
@@ -5,14 +5,22 @@ import PropTypes from 'prop-types'
...
+
+  if (isDev) {
+    console.warn(
```
Comment (sdalonzo) [inline packages/icons/src/Icon.js:11]: since `isDev`, maybe `console.trace` for a better stack trace?

Comment (craigpalermo) [reply packages/icons/src/Icon.js]: Somehow I didn't know `console.trace` was a thing 🤯

**Comment by sdalonzo** [comment]
@gbellini90 @gbellini911 this is the one i was referring to

**Comment by pcln-barry** [comment]
@craigpalermo @sdalonzo Just realized the builds don't seem to be running, allowing me to merge. Any way to kick them off manually? My usual cheat of an empty commit --amend didn't seem to work.

**Comment by pcln-barry** [comment]
Closing - will be re-opened soon to get those builds running

**Pull request closed without merging**